src/legacy/ui/public/filter_bar/query_filter.js

```diff
@@ -75,19 +85,14 @@ export function FilterBarQueryFilterProvider(Promise, indexPatterns, $rootScope,
    * @param {bool} global Whether the filter should be added to global state
    * @returns {Promise} filter map promise
    */
-  queryFilter.addFilters = function (filters, global) {
-
-    if (global === undefined) {
-      const configDefault = config.get('filters:pinnedByDefault');
-
-      if (configDefault === false || configDefault === true) {
-        global = configDefault;
-      }
+  queryFilter.addFilters = function (filters, addToGlobalState) {
+    if (addToGlobalState === undefined) {
+      addToGlobalState = pinnedByDefault;
```

Comment:
Can you use

```js
addToGlobalState = getNewPlatform().setup.core.uiSettings.get('filters:pinnedByDefault');
```

here? Instead of observable on line 40.

Reply:
Are we subscribing to an observable here because there isn't a guarantee the setting is available right away in new platform world? Otherwise (as @streamich points out) the observable probably isn't necessary in the first place.

Reply:
even if the settings is not available right away, using observable won't change behavior here.
i agree with @streamich to read the setting directly here, if we assume it might change. If we assume it doens't then just read it out on line 40, no need for observable.

Reply:
Went with @streamich suggestion.